elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -263,12 +306,319 @@ private Object fetchRelationship(Environment context) {
+                    idFieldName).get(), requestScope, entityClass, triplet.fieldName, triplet.parentResource);
+            toReturn.add(newParent);
+            Map<String, Object> relationshipsOnly = stripAttributes(triplet.input, entityClass,
+                    dictionary);
+            /* loop over relationships */
+            for(Map.Entry<String, Object> relationship : relationshipsOnly.entrySet()) {
+                Boolean isToOne = newParent.getRelationshipType(relationship.getKey()).isToOne();
+                List<Map<String, Object>> entryToAdd;
+                if(isToOne) {
+                    entryToAdd = new ArrayList<>();
+                    entryToAdd.add((Map<String, Object>) relationship.getValue());
+                } else {
+                    entryToAdd = (List) relationship.getValue();
+                }
+                /* loop over each resource of the relationship */
+                for(Map<String, Object> entry : entryToAdd) {
+                    toVisit.add(new Triplet(entry, Optional.of(newParent), relationship.getKey()));
+                }
+            }
+        }
+        return toReturn;
+    }
+
+    /**
+     * update the relationship between {@param parent} and the resource loaded by given {@param id}
+     * @param dataField data input
+     * @param id id
+     * @param requestScope request scope
+     * @param entityClass entity class
+     * @param fieldName graphql field name
+     * @param parent parent resource
+     * @return {@link PersistentResource} object
+     */
+    private PersistentResource updateRelationship(Map<String, Object> dataField, String id,
+                                            RequestScope requestScope, Class<?> entityClass,
+                                            String fieldName, Optional<PersistentResource> parent) {
+        PersistentResource resource = PersistentResource.loadRecord(entityClass, id, requestScope);
+        parent.ifPresent(persistentResource ->
+                persistentResource.updateRelation(fieldName, Collections.singleton(resource)));
+        return resource;
+    }
+
+    /**
+     * updates or creates existing/new entities
+     * @param dataField data input
+     * @param id id
+     * @param requestScope request scope
+     * @param fieldName field name
+     * @param parent parent resource
+     * @return {@link PersistentResource} object
+     */
+    private PersistentResource upsertObject(Map<String, Object> dataField, String id,
+                                             RequestScope requestScope, Class<?> entityClass,
+                                             String fieldName, Optional<PersistentResource> parent) {
+        Map<String, Object> input = stripRelationships(dataField, entityClass, requestScope.getDictionary());
+
+        Optional<PersistentResource> fetchEntry = fetchObject(requestScope, entityClass,
```

Comment:
What happened to the create part of the upsert?  If the object cannot be fetched it must be created.